src/Http/Client/Adapter/Curl.php

```diff
@@ -96,6 +96,10 @@ public function buildOptions(Request $request, array $options)
         if ($body) {
             $body->rewind();
             $out[CURLOPT_POSTFIELDS] = $body->getContents();
+            // GET requests with bodies require custom request to be used.
+            if (isset($out[CURLOPT_HTTPGET])) {
+                $out[CURLOPT_CUSTOMREQUEST] = 'get';
```

Comment:
I was wondering if just moving the CURLOPT_HTTPGET option to the end of the array would work.

However, this should work nicely because it doesn't matter if CURLOPT_HTTPGET is reset by curl.